pkg/backend/display/display.go

```diff
@@ -94,10 +90,21 @@ func startEventLogger(events <-chan engine.Event, done chan<- bool, path string)
 			apiEvent.Sequence, sequence = sequence, sequence+1
 			apiEvent.Timestamp = int(time.Now().Unix())
 
-			// If this is a diagnostic event, clean up the color directives from the emitted log.
-			if apiEvent.DiagnosticEvent != nil {
-				apiEvent.DiagnosticEvent.Message = cleanColorRenderingChars(apiEvent.DiagnosticEvent.Message)
-				apiEvent.DiagnosticEvent.Prefix = cleanColorRenderingChars(apiEvent.DiagnosticEvent.Prefix)
+			// If opts.Color == "never" (i.e. NO_COLOR is specified), clean up the color directives from the emitted logs.
+			if opts.Color == colors.Never {
+				if apiEvent.DiagnosticEvent != nil {
+					apiEvent.DiagnosticEvent.Message = colors.Never.Colorize(apiEvent.DiagnosticEvent.Message)
+					apiEvent.DiagnosticEvent.Prefix = colors.Never.Colorize(apiEvent.DiagnosticEvent.Prefix)
+					apiEvent.DiagnosticEvent.Color = "never"
+				}
+				if apiEvent.StdoutEvent != nil {
+					apiEvent.StdoutEvent.Message = colors.Never.Colorize(apiEvent.StdoutEvent.Message)
+					apiEvent.StdoutEvent.Color = "never"
+				}
+				if apiEvent.PolicyEvent != nil {
+					apiEvent.PolicyEvent.Message = colors.Never.Colorize(apiEvent.PolicyEvent.Message)
+					apiEvent.PolicyEvent.Color = "never"
+				}
```

Comment:
Nit: these cases are mutually exclusive, so this can be a `switch` statement

```suggestion
				switch {
				case apiEvent.DiagnosticEvent != nil:
					apiEvent.DiagnosticEvent.Message = colors.Never.Colorize(apiEvent.DiagnosticEvent.Message)
					apiEvent.DiagnosticEvent.Prefix = colors.Never.Colorize(apiEvent.DiagnosticEvent.Prefix)
					apiEvent.DiagnosticEvent.Color = "never"
				case apiEvent.StdoutEvent != nil:
					apiEvent.StdoutEvent.Message = colors.Never.Colorize(apiEvent.StdoutEvent.Message)
					apiEvent.StdoutEvent.Color = "never"
				case apiEvent.PolicyEvent != nil:
					apiEvent.PolicyEvent.Message = colors.Never.Colorize(apiEvent.PolicyEvent.Message)
					apiEvent.PolicyEvent.Color = "never"
				}
```